app/src/main/java/org/oppia/android/app/richtext/RteTextView.kt

```diff
@@ -0,0 +1,428 @@
+      }
+      drawableHeight = (drawableHeight.toDouble() * multipleFactor).toInt()
+      drawableWidth = (drawableWidth.toDouble() * multipleFactor).toInt()
+    }
+    val initialDrawableMargin = if (centerAlignImages) {
+      ((maxAvailableWidth - drawableWidth) / 2).coerceAtLeast(0)
+    } else {
+      0
+    }
+
+    val newDrawableBounds = Rect(
+      /* left= */ initialDrawableMargin,
+      /* top= */ 0,
+      /* right= */ drawableWidth + initialDrawableMargin,
+      /* bottom= */ drawableHeight
+    )
+    if (newDrawableBounds != drawable.bounds) {
+      drawable.bounds = newDrawableBounds
+      return true // New bounds have been computed.
+    }
+    // The drawable doesn't need to be redrawn since it's the same. Note that this could result in a
+    // new drawable not being drawn, but that's not expected since the default bounds is
+    // (0, 0 - 0, 0) which, if computed here, won't lead to anything being drawn, anyway.
+    return false
+  }
+
+  private fun getImageDrawables(): List<Drawable> {
+    val imageSpans = (text as? Spanned)?.getSpans(0, text.length, ImageSpan::class.java)
+    return imageSpans?.map(ImageSpan::getDrawable) ?: listOf()
+  }
+
+  /**
+   * If raw HTML was previously bound, rebinds it. This should be used in situations when settings
+   * have changed that may lead to the HTML being interpreted differently.
+   *
+   * Calling this method should guarantee all image drawable bounds are recomputed.
+   */
+  private fun reBindHtml() {
+    boundRawHtml?.let { text = it }
+  }
+
+  /**
+   * A custom [LifecycleOwner] that hackily superimposes a view's lifecycle into the buckets of an
+   * activity's lifecycle. This is done privately since it should only ever be used for LiveData
+   * observers whose results can be processed within the window this owner classifies as "resumed"
+   * in activity terminology (view is attached).
+   *
+   * One limitation is that this class treats view detachment as a destruction. Views can be
+   * detached/reattached in legitimate view hierarchy organizations. This class works around this
+   * limitation by returning a new lifecycle after detachment that's again in the initialized state.
+   * If the view gets attached again, new observers should correctly be processed. Observers must
+   * not be long-lived, and will need to reset themselves to work against this situation. To
+   * simplify management, callers should just use [observeOnce].
+   *
+   * Note: the internal lifecycle events processed by this class SHOULD NOT BE CONSTRUED to mean
+   * their corresponding value (e.g. a state of RESUMED does NOT mean that the fragment/activity to
+   * which this view is bound to, if any, is in a RESUMED state--that is not guaranteed nor can it
+   * be).
+   *
+   * Reference for this implementation:
+   * https://developer.android.com/topic/libraries/architecture/lifecycle#implementing-lco.
+   */
+  private class ViewLifecycleOwner: LifecycleOwner {
```

Comment:
Let's move this to class to a separate file, could be useful for other custom views in the future?

Reply:
Per the warnings here, I don't think we actually want to do that though. Views are intentionally not compatible with AndroidX lifecycle, and this approach bakes in specific assumptions that lead to it needing to be used in a very specific way to avoid subtle issues or leaks.